code/game/objects/structures/displaycase.dm

```diff
@@ -114,7 +126,30 @@
 	src.icon = I
 	return
 
-/obj/structure/displaycase/attackby(obj/item/weapon/W, mob/user, params)
+/obj/structure/displaycase/attackby(obj/item/W, mob/user, params)
+	if(is_trophy)
+		if(is_type_in_typecache(W, blacklisted_cargo_types))
+			user << "You think putting [W] in would be a bad idea."
+			return
+		var/chosen_plaque = stripped_input(user, "What would you like the plaque to say?", "Trophy Plaque")

```

Comment:
this is a sleep, check the item is not qdeleted and or already has an item inside it after this.